src/code/z_camera.c

```diff
@@ -1,3070 +1,6778 @@
+    } else {
+        // rotate yaw to follow player.
+        eyeAdjustment.yaw =
+            Camera_CalcDefaultYaw(camera, atEyeNextGeo.yaw, camera->playerPosRot.rot.y, norm1->unk_14, sp94);
+        eyeAdjustment.pitch =
+            Camera_CalcDefaultPitch(camera, atEyeNextGeo.pitch, norm1->pitchTarget, anim->slopePitchAdj);
+    }
 
-    D_8011D3A0.w = val;
-    if (val & 1) {
-        camera->unk_14C |= 0x400;
+    // set eyeAdjustment pitch from 79.65 degrees to -85 degrees
+    if (eyeAdjustment.pitch > 0x38A4) {
+        eyeAdjustment.pitch = 0x38A4;
     }
-    if (val & 2) {
-        camera->unk_14C |= 0x10;
+    if (eyeAdjustment.pitch < -0x3C8C) {
+        eyeAdjustment.pitch = -0x3C8C;
     }
-}
-
-s32 Camera_Parallel4(Camera* camera) {
-    return func_80047394(camera);
-}
-
-s32 Camera_Parallel0(Camera* camera) {
-    return func_80047394(camera);
-}
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/Camera_Jump1.s")
+    Camera_Vec3fVecSphGeoAdd(eyeNext, at, &eyeAdjustment);
+    if ((camera->status == CAM_STAT_ACTIVE) && (!(norm1->interfaceFlags & 0x10))) {
+        anim->swingYawTarget = BINANG_ROT180(camera->playerPosRot.rot.y);
+        if (anim->startSwingTimer > 0) {
+            func_80046E20(camera, &eyeAdjustment, norm1->distMin, norm1->unk_0C, &sp98, &anim->swing);
+        } else {
+            sp88 = *eyeNext;
+            anim->swing.swingUpdateRate = camera->yawUpdateRateInv = norm1->unk_0C * 2.0f;
+            if (Camera_BGCheck(camera, at, &sp88)) {
+                anim->swingYawTarget = atEyeNextGeo.yaw;
+                anim->startSwingTimer = -1;
+            } else {
+                *eye = *eyeNext;
+            }
+            anim->swing.unk_18 = 0;
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/Camera_Jump2.s")
+        if (anim->swing.unk_18 != 0) {
+            camera->inputDir.y = Camera_LERPCeilS(
+                camera->inputDir.y + BINANG_SUB(BINANG_ROT180(anim->swing.unk_16), camera->inputDir.y),
+                camera->inputDir.y, 1.0f - (0.99f * sp98), 0xA);
+        }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_camera/Camera_Jump3.s")
+        if (norm1->interfaceFlags & 4) {
+            camera->inputDir.x = -atEyeGeo.pitch;
+            camera->inputDir.y = BINANG_ROT180(atEyeGeo.yaw);
+            camera->inputDir.z = 0;
+        } else {
+            OLib_Vec3fDiffToVecSphGeo(&eyeAdjustment, eye, at);
+            camera->inputDir.x = eyeAdjustment.pitch;
+            camera->inputDir.y = eyeAdjustment.yaw;
+            camera->inputDir.z = 0;
+        }
 
-s32 Camera_Jump4(Camera* camera) {
-    return func_80047394(camera);
-}
+        // crit wiggle
+        if (gSaveContext.health <= 0x10 && ((camera->globalCtx->state.frames % 256) == 0)) {
+            t2 = Math_Rand_ZeroOne() * 10000.0f;
+            camera->inputDir.y = t2 + camera->inputDir.y;
+        }
```

Comment:
i didnt know this was in camera... lmao
also health in decimal pls
t2 can be called randOffset? wiggleAmount idk